tests/test_libzim.py

```diff
@@ -437,3 +443,155 @@ def get_contentProvider(self):
+    zim_path = tmpdir / "test.zim"
+    path = "path"
+    fpath = "welcome.txt"
+    content = "hello world"
+
+    class FileItem:
+        def __init__(self, path, fpath):
+            self.path = path
+            self.cp = FileProvider(fpath)
+
+        def get_path(self):
+            return self.path
+
+        def get_title(self):
+            return ""
+
+        def get_mimetype(self):
+            return "text/plain"
+
+        def get_contentProvider(self):
+            return self.cp
+
+    with open(fpath, "w", encoding="UTF-8") as fh:
+        fh.write(content)
+
+    with Creator(zim_path) as zim_creator:
+        zim_creator.add_item(FileItem(path, fpath))
+
+    archive = Archive(zim_path)
+    assert bytes(archive.get_entry_by_path(path).get_item().content).decode("UTF-8") == content
+
+    # manual FileProvider calls for coverage
+    fp = FileProvider(fpath)
+    assert Blob(content.encode("UTF-8")).size() == fp.feed().size()
+    # single call exhausts it as under 1MiB
+    assert Blob(b"").size() == fp.feed().size()
+
+
+def test_fileprovider_largefile(tmpdir):
+    zim_path = tmpdir / "test.zim"
+    path = "path"
+    fpath = "somebin"
+    buffsize = 1024 * 1024
+    # create a dummy file twice the size of the buffer
+    with open(fpath, "wb") as fh:
+        for _ in range(0, buffsize // 1024):
+            fh.write(b"ab" * 1024)
+    fsize = os.path.getsize(fpath)
+
+    class FileItem:
+        def __init__(self, path, fpath):
+            self.path = path
+            self.cp = FileProvider(fpath)
+
+        def get_path(self):
+            return self.path
+
+        def get_title(self):
+            return ""
+
+        def get_mimetype(self):
+            return "text/plain"
+
+        def get_contentProvider(self):
+            return self.cp
+
+    with Creator(zim_path) as zim_creator:
+        zim_creator.add_item(FileItem(path, fpath))
+
+    archive = Archive(zim_path)
+    assert archive.get_entry_by_path(path).get_item().content.nbytes == fsize
+
+    # manual FileProvider calls for coverage
+    fp = FileProvider(fpath)
+    # we need two calls to exhaust this (uses 1MiB buffer)
+    assert fp.feed().size() == buffsize
+    assert fp.feed().size() == buffsize
+    assert Blob(b"").size() == fp.feed().size()
+
+
+def test_fileprovider_fails_missingfile(tmpdir):
+    zim_path = tmpdir / "test.zim"
+    path = "path"
+    fpath = "missing-file"
+
+    class FileItem:
+        def __init__(self, path, fpath):
+            self.path = path
+            self.cp = FileProvider(fpath)
+
+        def get_path(self):
+            return self.path
+
+        def get_title(self):
+            return ""
+
+        def get_mimetype(self):
+            return "application/octet-stream"
+
+        def get_contentProvider(self):
+            return self.cp
+
+    with Creator(zim_path) as zim_creator:
+        with pytest.raises(IOError):
+            zim_creator.add_item(FileItem(path, fpath))
```

Comment:
Well, if we move the gen_blob using a context manager, we will catch the error when libzim want to get the content.
By opening it the constructor, we catch the error when we create the `FileProvile`...

Reply:
No, the error is raised by the getsize call in the constructor, regardless of where the file is opened.